cscs-checks/libraries/math/trilinos_compile_run.py

```diff
@@ -62,3 +63,12 @@ def cdt2006_workaround_intel(self):
                     'ln -s $CRAY_NETCDF_HDF5PARALLEL_PREFIX/lib/pkgconfig/'
                     'netcdf-cxx4_parallel.pc netcdf_c++4_parallel.pc']
                 self.variables['PKG_CONFIG_PATH'] = '.:$PKG_CONFIG_PATH'
+
+    @rfm.run_before('compile')
+    def cdt2006_workaround_dynamic(self):
+        if (os_ext.cray_cdt_version() == '20.06' and
+            self.linkage == 'dynamic'):
+            self.variables['PATH'] = '/opt/cray/pe/cce/10.0.1/cce-clang/x86_64/bin:$PATH'
+            self.prgenv_flags[self.current_environ.name] += ['-fuse-ld=lld']
```

Comment:
This workaround is only for `PrgEnv-gnu` and I would suggest to put in another function that would be run before compile also.